query-creator/create-query/steps/aws.js

```diff
@@ -361,6 +368,36 @@ exports.generateLogSQL = (queryId, runnableUrl, authDomain, sourceKey) => {
   `;
 };
 
+/*
+Generates a very wide row including all fields from the user log.
+*/
+exports.generateUserLogSQL = (usernames, activities, start_date, end_date) => {
+  const logDb = process.env.LOG_ATHENA_DB_NAME;
+  const where = [];
+  if (usernames.length > 0) {
+    where.push(`log.username IN (${usernames.map(u => `'${escapeSingleQuote(u)}'`).join(", ")})`)
```

Comment:
usernames might be large enough that we have to handle them like learner: upload json to S3 and do a join with it.

But I think it is good to start this more simple way and we can see if we run into problems.